components/core/src/metrics.rs

```diff
@@ -37,27 +36,47 @@ enum Operation {
     Increment,
 }
 
-lazy_static! {
-    static ref SEND_CHANNEL : Mutex<Sender<(String, Operation)>> = {
-        let (tx, rx) : (Sender<(String, Operation)>, Receiver<(String, Operation)>) = channel();
-        let mut statsd_client = statsd_client();
-        thread::spawn(move || {
-            process_receives(rx, &mut statsd_client)
+// Helper type
+type CounterOp = (String, Operation);
+
+use std::sync::{Once, ONCE_INIT};
+
+static mut SENDER: *const Sender<CounterOp> = 0 as *const Sender<CounterOp>;
+
+static INIT: Once = ONCE_INIT;
+
+fn get_sender() -> Sender<CounterOp> {
+    unsafe {
+        INIT.call_once(|| {
+            SENDER = Box::into_raw(Box::new(do_init()));
         });
-        Mutex::new(tx)
-    };
+        (*SENDER).clone()
+    }
+}
+
+fn do_init() -> Sender<CounterOp> {
+    let (tx, rx): (Sender<CounterOp>, Receiver<CounterOp>) = channel();
+    let mut statsd_client = statsd_client();
+    thread::spawn(move || process_receives(rx, &mut statsd_client));
+    tx
 }
 
-fn process_receives(rx: Receiver<(String, Operation)>, statsd_client: &mut Option<Client>) {
+fn process_receives(rx: Receiver<CounterOp>, statsd_client: &mut Option<Client>) {
     loop {
         let (counter, op): (String, Operation) = rx.recv().unwrap();
```

Comment:
You'll want to match on `recv()` here and handle failure cases (logging them at the very least). Unwrapping here will result in the entire supervisor crashing if there was a problem with the channel